Make singleton html and handlebar tags (like partials) use 4 space indentation

adnrs96 opened the pull request:
Make singleton html and handlebar tags (like partials) use 4 space indentation.

adnrs96 commented:
To see what kind of changes this brings to linter, checkout the test case additions for pretty_print_html. Those cases should make it pretty clear what kind of changes one should expect in linting from these changes.

timabbott commented:
I think for the partials, we definitely want 2-space indentation.  It might be a good idea to use 2-space indentation for the other HTML singleton tags as well, since it'll distinguish them from just something inside a block (I found that a bit confusing when reading the first commit).

adnrs96 commented:
Hi @timabbott 
So I addressed your comments with the updates to the commits. I didn't squash the new commits completely but arranged them meaningfully so that it might help you in reviewing. 
Also I added another commit with achieves two things:
* Reduces the complexity of a if else statements (not really a requirement but just to state why I even went down the road which resulted in this commit)
* Enforces 4 space indents on tags in situations like:
 ```
<button type="button"
class="btn btn-primary btn-small">{{t "Yes" }}</button>
```
Currently we just honour whatever indentation the developer provided to the second line (the continuation line of the button tag) with respect to its origin line. The last commit starts to enforce a 4 space indent there as well. I separated it out so that it can be merged if that will improve the state of our linter.

timabbott commented:
Merged, after squashing and cleaning up the commit messages to consistently refer to 2 spaces.  Thanks @adnrs96!

timabbott commented:
@shubhamdhama FYI :)
